python/graphstorm/gconstruct/transform.py

```diff
@@ -935,6 +935,143 @@ def call(self, feats):
                 f"The feature {self.feat_name} has to be integers or floats."
         return {self.feat_name: feats}
 
+class HardEdgeNegativeTransform(TwoPhaseFeatTransform):
+    """ Translate input data into node ids for hard negative stored as edge features
```

Comment:
do i understand correctly that users prepare the hard negative samples and stores them as edge features? this class interprets the edge features and loads them as hard negative edges.

Reply:
are these negative edges stored in the DGL graph as edge features? the data loader needs to understand the edge features.